<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>540</width>
    <height>231</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Криптоалгоритмы</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>541</width>
      <height>231</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="lab1">
     <attribute name="title">
      <string>ЛР1</string>
     </attribute>
     <widget class="QPushButton" name="pushEncrypt">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Зашифровать</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushDecrypt">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>170</y>
        <width>101</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Расшифровать</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="editSource">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="editEncrypted">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Исходная строка</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Зашифрованная строка</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>0</y>
        <width>221</width>
        <height>201</height>
       </rect>
      </property>
      <property name="title">
       <string>Криптоалгоритм</string>
      </property>
      <widget class="QRadioButton" name="rbtnSingleReshuffle">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>161</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Одиночная перестановка</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnSingleWKey">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>211</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Одиночная перестановка с ключем</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnDoubleReshuffle">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>141</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Двойная перестановка</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnCaesar">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>91</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Шифр Цезаря</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnGronsfeld">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>121</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Шифр Гронсвельда</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnManyAlph">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>161</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Многоалфавитная замена</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnGamma">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>101</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Гаммирование</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Ключ</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="editKey">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="lab2">
     <attribute name="title">
      <string>ЛР2</string>
     </attribute>
     <widget class="QLineEdit" name="editAsymmSource">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>391</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="editAsymmPubKey">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>391</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="editAsymmPrivateKey">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>391</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="editAsymmEncrypted">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>391</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Закрытый ключ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Открытый ключ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Зашифрованная строка</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Исходная строка</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushAsymmEncrypt">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>150</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Зашифровать</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushAsymmDecrypt">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>180</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Расшифровать</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>10</y>
        <width>120</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>Криптоалгоритм</string>
      </property>
      <widget class="QRadioButton" name="rbtnElGamal">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Эль-Гамаль</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="rbtnRsa">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>RSA</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pushGenKey">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>120</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Сгенерировать ключ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Генерация ключа</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="editKeySeed">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>391</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
